### PR TITLE
Arm/poa cleanup

### PR DESCRIPTION
**Pull request by arminsabouri**
Cleaning up double code in task. Mainly around inserting and persisting new blocks and calling fork choice update.
There are tons of warnings after the upstream rebase. I tried to squash as many as I can here related to consensus. I'll grab the other ones in a future commit. Hopefully soon the lint job on gh will pass 💯 

Note: I separated out the refactor commits from the warning fix commits for ease of review

### CODE REVIEW

File: crates/consensus/common/src/utils.rs [hunk 1]
```diff
@@ -46,15 +49,21 @@ pub fn read_staker_balance(
 }
 
 #[derive(Debug)]
+/// Error that can occur while recovering the authority list
```
Comment (BatiGencho) [inline crates/consensus/common/src/utils.rs:51]: Dont we want to use thiserror here or displaydoc with macros for better error message handling, esp. with nested errors ?

File: crates/consensus/authority/src/task.rs [hunk 2]
```diff
@@ -102,7 +114,9 @@ where
             pipe_line_events: None,
             btc_server,
             bitcoin_block_header,
-            bitcoin_block_source_address,
+            bitcoin_block_source: MempoolSpace::new(
+                bitcoin_block_source_address.clone().to_string(),
```
Comment (BatiGencho) [inline crates/consensus/authority/src/task.rs]: do we need to clone it here, the to_string() should construct and return a new owned String right ?

File: crates/consensus/authority/src/task.rs [hunk 3]
```diff
@@ -171,92 +182,14 @@ where
                                             // TODO (armins) remove expect
                                             .expect("block is valid");
```
Comment (BatiGencho) [inline crates/consensus/authority/src/task.rs]: do we want to handle such expects ?

File: crates/consensus/common/src/utils.rs [hunk 4]
```diff
@@ -76,12 +85,15 @@ pub fn recovery_authority(header: &Header) -> Result<secp256k1::PublicKey, Recov
 }
 
 #[derive(Debug)]
+/// Errors that can occur while reading the authority list from the block header
 pub enum GetAuthoritiesError {
+    /// Failed to deserialize the extra data
```
Comment (BatiGencho) [inline crates/consensus/common/src/utils.rs:89]: same comment here reg. the error handling - displaydoc / thiserror ?

File: crates/consensus/common/src/utils.rs (hunk 1 above)
Comment (arminsabouri) [reply crates/consensus/common/src/utils.rs:51]: You have more Rust experience I'll follow your recommendation. What do you think is the best way to move forward?

File: crates/consensus/authority/src/task.rs (hunk 3 above)
Comment (arminsabouri) [reply crates/consensus/authority/src/task.rs]: Good point! Added error handling to this

File: crates/consensus/authority/src/task.rs (hunk 2 above)
Comment (arminsabouri) [reply crates/consensus/authority/src/task.rs]: Good catch! Fixed [22ba968](https://github.com/botanix-labs/Macbeth/pull/59/commits/22ba968aa10db41d2a45868c8f5b57bb02569fd2)

**Review by scottmillner — APPROVED** [review]

File: crates/consensus/common/src/utils.rs (hunk 1 above)
Comment (arminsabouri) [reply crates/consensus/common/src/utils.rs:51]: @BatiGencho Im going to merge the clean up. But want to make sure this is addresses. Lmk what the best path going forward is and I can make a change in a future PR

Comment (BatiGencho) [reply crates/consensus/common/src/utils.rs:51]: I went through it, seems clean to me. Good that you merged it!

File: crates/consensus/authority/src/task.rs (hunk 3 above)
Comment (BatiGencho) [reply crates/consensus/authority/src/task.rs]: Great!

File: crates/consensus/authority/src/task.rs (hunk 2 above)
Comment (BatiGencho) [reply crates/consensus/authority/src/task.rs]: Great!

**Pull request merged**